WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewController.swift

```diff
@@ -27,23 +27,58 @@ final class CardReaderSettingsConnectedViewController: UIViewController, CardRea
             DDLogError("Unexpectedly unable to downcast to CardReaderSettingsConnectedViewModel")
             return
         }
+
+        self.viewModel?.didUpdate = onViewModelDidUpdate
     }
 
     // MARK: - Overridden Methods
 
     override func viewDidLoad() {
         super.viewDidLoad()
-
         registerTableViewCells()
         configureNavigation()
         configureSections()
         configureTable()
     }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        checkForCardReaderUpdate()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        viewModel?.didUpdate = nil
+        super.viewWillDisappear(animated)
+    }
 }
 
 // MARK: - View Configuration
 //
 private extension CardReaderSettingsConnectedViewController {
+    func checkForCardReaderUpdate() {
+        guard ServiceLocator.featureFlagService.isFeatureFlagEnabled(.cardPresentSoftwareUpdates) else {
+            return
+        }
+        guard let viewModel = viewModel else {
```

Comment:
I was going to say that both guards could be collapsed into one, but leaving the check for the feature flag separated would make removing it easier so 🤷🏼